process/sync/storageBootstrap/metaStorageBootstrapper.go

```diff
@@ -118,3 +121,8 @@ func (msb *metaStorageBootstrapper) cleanupNotarizedStorage(metaBlockHash []byte
 		}
 	}
 }
+
+func (msb *metaStorageBootstrapper) applySelfNotarizedHeaders(selfNotarizedHeadersHashes [][]byte) ([]data.HeaderHandler, error) {
+	selfNotarizedHeaders := make([]data.HeaderHandler, 0, len(selfNotarizedHeadersHashes))
```

Comment:
Yes, it set headers which are notarized from your chain in the cross chain. If you are a shard you need to know which are your last header notarized by meta because it gives you your final. But for meta is not so important what shards notarized from you as it does not affect you. Metachain makes the rules :). Made a 0 size empty list